pkg/dynamicconfig/table.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamicconfig
+
+import (
+	"sort"
+	"strconv"
+
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/index"
+
+	"github.com/cilium/cilium/pkg/k8s"
+)
+
+const TableName = "cilium-configs"
+
+var (
+	keyIndex = statedb.Index[DynamicConfig, Key]{
+		Name: "config-name-source-key",
```

Comment:
`Name` here and below out of date. I'd just have `key` here, and maybe `name` below (these aren't actually yet used for anything, but eventually I think we'll have some command or documentation for all the tables and their indexes...).